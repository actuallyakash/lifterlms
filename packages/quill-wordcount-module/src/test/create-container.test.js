import createContainer from '../create-container';

describe( 'createContainer()', () => {
	const testData = [
		[ 'No min and no max', { min: null, max: null } ],
		[ 'Min and no max', { min: 5, max: null, l10n: { min: 'Min' } } ],
		[ 'Max and no min', { min: null, max: 1000, l10n: { max: 'Max' } } ],
		[ 'Max and min', { min: 1, max: 2, l10n: { min: 'Min', max: 'Max' } } ],
	];
	test.each( testData )( '%s', ( testName, opts ) => {
		expect( createContainer( opts ) ).toMatchSnapshot();
	} );
} );
