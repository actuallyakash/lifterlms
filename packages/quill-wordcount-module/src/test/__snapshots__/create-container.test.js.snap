// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createContainer() Max and min 1`] = `
<div
  class="ql-wordcount ql-toolbar ql-snow"
  style="margin-top: -1px; font-size: 85%;"
>
  <i
    class="ql-wordcount-min"
    style="opacity: 0.5; margin-right: 10px;"
  >
    Min: 1
  </i>
  <i
    class="ql-wordcount-max"
    style="opacity: 0.5; margin-right: 10px;"
  >
    Max: 2
  </i>
</div>
`;

exports[`createContainer() Max and no min 1`] = `
<div
  class="ql-wordcount ql-toolbar ql-snow"
  style="margin-top: -1px; font-size: 85%;"
>
  <i
    class="ql-wordcount-max"
    style="opacity: 0.5; margin-right: 10px;"
  >
    Max: 1,000
  </i>
</div>
`;

exports[`createContainer() Min and no max 1`] = `
<div
  class="ql-wordcount ql-toolbar ql-snow"
  style="margin-top: -1px; font-size: 85%;"
>
  <i
    class="ql-wordcount-min"
    style="opacity: 0.5; margin-right: 10px;"
  >
    Min: 5
  </i>
</div>
`;

exports[`createContainer() No min and no max 1`] = `
<div
  class="ql-wordcount ql-toolbar ql-snow"
  style="margin-top: -1px; font-size: 85%;"
/>
`;
